Mathlib/CategoryTheory/FiberedCategory/BasedCategory.lean

```diff
@@ -0,0 +1,259 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Lezeau, Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.FiberedCategory.HomLift
+import Mathlib.CategoryTheory.Bicategory.Strict
+import Mathlib.CategoryTheory.Functor.Category
+
+/-!
+# The bicategory of based categories
+
+In this file we define the type `BasedCategory 𝒮`, and give it the structure of a strict
+bicategory. Given a category `𝒮`, we define the type `BasedCategory 𝒮` as the type of categories
+`𝒳` equiped with a functor `𝒳.p : 𝒳 ⥤ 𝒮`.
+
+We also define functors between based categories `𝒳 𝒴 : BasedCategory 𝒮`, via the structure
+`BasedFunctor 𝒳 𝒴`. These are defined as functors between the underlying categories `𝒳.cat` and
+`𝒴.cat` which commute with the projections to `𝒮`.
+
+Natural transformations between based functors `F G : BasedFunctor 𝒳 𝒴` are given by the structure
+`BasedNatTrans F G`. These are defined as natural transformations `α` between the functors
+underlying `F` and `G` such that `α.app a` lifts `𝟙 S` whenever `𝒳.p.obj a = S`.
+-/
+
+universe u₁ v₁
+
+open CategoryTheory Functor Category NatTrans IsHomLift
+
+variable {𝒮 : Type u₁} [Category.{v₁} 𝒮]
+
+/-- A based category over `𝒮` is a category `𝒳` together with a functor `p : 𝒳 ⥤ 𝒮` -/
+structure BasedCategory (𝒮 : Type u₁) [Category.{v₁} 𝒮] where
+  /-- The type of objects in a `BasedCategory`-/
+  cat : Type u₁
```

Comment:
I think it is important to allow different universe parameters for the categories above and below.

Reply:
This is what I had originally, but if I let `cat` have type `u₂` then the linter complains that `u₂` and `u₁` always appear together.

Reply:
I believe this is a case where it is ok to insert `@[nolint checkUnivs]` before the declaration. Then, the `universe` declaration at the beginning of the file should be `universe v₂ u₂ v₁ u₁` (with `cat : Type u₂` and `[... : Category.{v₂} cat]`. Then, we may use `BasedCategory.{v₂, u₂} 𝒮` in order to refer explicitly to the category of based caetgory with certain universe parameters.